packages/wrangler/src/pages/dev.ts

```diff
@@ -476,15 +521,80 @@ export const Handler = async (args: PagesDevArguments) => {
 					routesModule,
 					defineNavigatorUserAgent,
 				});
+
+				for (const dep in bundle.dependencies) {
+					const resolvedDep = resolve(functionsDirectory, dep);
+					/*
+					 * EXCLUDE:
+					 *   - the "/functions" directory because we're already watching it
+					 *   - everything in "./.wrangler", as it's mostly cache and
+					 *     temporary files
+					 *   - any bundle dependencies we are already watching
+					 *   - anything outside of the current working directory, since we
+					 *     are expecting `wrangler pages dev` to be run from the Pages
+					 *     project root folder
+					 */
+					if (
+						!resolvedDep.match(/\/functions\/./) &&
+						!resolvedDep.match(/\.wrangler\/./) &&
```

Comment:
Does this affect middleware/templates? I think we sometimes put those in `.wrangler`?

Reply:
yes it indeed does. those will be filtered out. It *seemed* like that would be OK, since technically that's not something users should tamper with? In my head the files that'd need to be watched here, are user modified files (with "modified" being add/change/delete). Furthermore, but I'd have to dbl check this again, I seem to remember that not filtering that `.wrangler/tmp` folder would result in some circular dependency situation? where chokidar would enter a perpetual state of triggering change events

Reply:
That makes sense! Users should ideally not be modifying these files directly